File: exemples/js/04_query_payment.js
/**
 * Example: Query payment status
 *
 * This example shows how to:
 * 1. Check if a payment has been processed
 * 2. Query payment status for different operators
 */

import {
  initTronWeb,
  getPaymentSplitterContract,
  printResult
} from './utils.js';

async function queryPaymentStatus(splitter, operatorAddress, paymentId) {
  const isProcessed = await splitter.isPaymentProcessed(operatorAddress, paymentId).call();

  return {
    operator: operatorAddress,
    paymentId,
    isProcessed
  };
}

async function main() {
  try {
    console.log('=== Query Payment Status Example ===\n');

    // Initialize TronWeb
    const tronWeb = initTronWeb();
    console.log('Connected to Shasta testnet\n');

    // Load PaymentSplitter contract
    const splitter = await getPaymentSplitterContract(tronWeb);
    console.log('PaymentSplitter contract loaded\n');

    // Example payment ID (replace with actual payment ID from a previous transaction)
    const operatorAddress = tronWeb.defaultAddress.base58;
    const examplePaymentId = '0x00000000000000000000000000000000'; // Replace with real payment ID

    console.log('Query Parameters:');
    console.log(`  Operator: ${operatorAddress}`);
    console.log(`  Payment ID: ${examplePaymentId}\n`);

    // Query payment status
    console.log('Querying payment status...');
    const paymentStatus = await queryPaymentStatus(splitter, operatorAddress, examplePaymentId);

    printResult('Payment Status', paymentStatus);

    if (paymentStatus.isProcessed) {
      console.log('✓ Payment has been processed\n');
    } else {
      console.log('✗ Payment has not been processed yet\n');
    }

  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

// Run the example
main();

File: exemples/ruby/04_query_payment.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Example: Query payment status
#
# This example shows how to:
# 1. Check if a payment has been processed
# 2. Query payment status for different operators
#
# IMPORTANT: Run with bundle exec to avoid gem conflicts:
#   cd examples && bundle exec ruby ruby/04_query_payment.rb

require_relative 'utils'

def query_payment_status(client, operator_address, payment_id)
  contract_service = client.contract_service
  is_processed = contract_service.payment_processed?(
    PAYMENT_SPLITTER_ADDRESS,
    operator_address,
    payment_id
  )

  {
    'Operator' => operator_address,
    'Payment ID' => payment_id,
    'Is Processed' => is_processed
  }
end

def main
  puts '=== Query Payment Status Example ==='
  puts

  # Initialize Tron client
  client = init_tron_client
  puts 'Connected to Shasta testnet'
  puts

  operator_private_key = ENV['OPERATOR_PRIVATE_KEY']
  unless operator_private_key
    puts '❌ Error: OPERATOR_PRIVATE_KEY not found in .env file'
    exit 1
  end

  operator_address = get_address_from_private_key(operator_private_key)

  # Example payment ID (replace with actual payment ID from a previous transaction)
  example_payment_id = '0x00000000000000000000000000000000' # Replace with real payment ID

  puts 'Query Parameters:'
  puts "  Operator: #{operator_address}"
  puts "  Payment ID: #{example_payment_id}"
  puts

  begin
    # Query payment status
    puts 'Querying payment status...'
    payment_status = query_payment_status(client, operator_address, example_payment_id)

    print_result('Payment Status', payment_status)

    if payment_status['Is Processed']
      puts '✓ Payment has been processed'
    else
      puts '✗ Payment has not been processed yet'
    end
    puts

  rescue => e
    puts "❌ Error: #{e.message}"
    puts "Backtrace: #{e.backtrace.first(3).join("\n")}" if ENV['DEBUG']
    exit 1
  end
end

main if __FILE__ == $PROGRAM_NAME
